From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lordofpipes <lordpipe@protonmail.com>
Date: Fri, 17 Jan 2025 06:01:34 -0700
Subject: [PATCH] Enforce insecure profile


diff --git a/net/minecraft/network/protocol/game/ClientboundPlayerChatPacket.java b/net/minecraft/network/protocol/game/ClientboundPlayerChatPacket.java
index 3ee2510f31cd8f09883d48fcc7ab502c3614ee7f..4af139cf1af728a0071f0972e8cfc2269537ab8e 100644
--- a/net/minecraft/network/protocol/game/ClientboundPlayerChatPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundPlayerChatPacket.java
@@ -45,7 +45,7 @@ public record ClientboundPlayerChatPacket(
         buffer.writeVarInt(this.globalIndex);
         buffer.writeUUID(this.sender);
         buffer.writeVarInt(this.index);
-        buffer.writeNullable(this.signature, MessageSignature::write);
+        buffer.writeNullable(null, MessageSignature::write);
         this.body.write(buffer);
         FriendlyByteBuf.writeNullable(buffer, this.unsignedContent, ComponentSerialization.TRUSTED_STREAM_CODEC);
         FilterMask.write(buffer, this.filterMask);
diff --git a/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java b/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java
index 8212337ab7c30491926f50a9a9fa1f0451c06b47..8aba81bf1b296ae8b039720b35ed4bc92d064ff2 100644
--- a/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java
@@ -174,7 +174,7 @@ public class ClientboundPlayerInfoUpdatePacket implements Packet<ClientGamePacke
                 if (chatSession != null && chatSession.profilePublicKey().hasExpired()) {
                     chatSession = null;
                 }
-                buffer.writeNullable(chatSession, RemoteChatSession.Data::write);
+                buffer.writeNullable(null, RemoteChatSession.Data::write);
             }
             // Paper end - Prevent causing expired keys from impacting new joins
         ),
