From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lordofpipes <lordpipe@protonmail.com>
Date: Fri, 17 Jan 2025 06:01:34 -0700
Subject: [PATCH] Enforce insecure profile


diff --git a/net/minecraft/network/protocol/game/ClientboundPlayerChatPacket.java b/net/minecraft/network/protocol/game/ClientboundPlayerChatPacket.java
index 7cd03722e45631ee8f1118dddf464510343313ed..75caebb3b8ce35f43c1500f6c14df4bdd07f7770 100644
--- a/net/minecraft/network/protocol/game/ClientboundPlayerChatPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundPlayerChatPacket.java
@@ -45,7 +45,7 @@ public record ClientboundPlayerChatPacket(
         output.writeVarInt(this.globalIndex);
         output.writeUUID(this.sender);
         output.writeVarInt(this.index);
-        output.writeNullable(this.signature, MessageSignature::write);
+        output.writeNullable(null, MessageSignature::write);
         this.body.write(output);
         FriendlyByteBuf.writeNullable(output, this.unsignedContent, ComponentSerialization.TRUSTED_STREAM_CODEC);
         FilterMask.write(output, this.filterMask);
diff --git a/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java b/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java
index aaa497096f07f2ef9da62b9760f5c9aa0b0263d3..d319a229296496869f34e138332837ef11c5f15e 100644
--- a/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java
@@ -174,7 +174,7 @@ public class ClientboundPlayerInfoUpdatePacket implements Packet<ClientGamePacke
                 if (chatSession != null && chatSession.profilePublicKey().hasExpired()) {
                     chatSession = null;
                 }
-                output.writeNullable(chatSession, RemoteChatSession.Data::write);
+                output.writeNullable(null, RemoteChatSession.Data::write);
             }
             // Paper end - Prevent causing expired keys from impacting new joins
         ),
