From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lordpipe <lordpipe@protonmail.com>
Date: Mon, 12 Dec 2022 01:22:32 -0700
Subject: [PATCH] Enforce insecure profile


diff --git a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java
index 07df3299f1d1aa5506e1f6f146347d53e0278d9c..5709ccd352d75a984af7e4de7055bc589429fb37 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java
@@ -11,19 +11,31 @@ import net.minecraft.network.protocol.PacketType;
 
 public record ServerboundChatPacket(String message, Instant timeStamp, long salt, @Nullable MessageSignature signature, LastSeenMessages.Update lastSeenMessages)
     implements Packet<ServerGamePacketListener> {
+    // Enforce insecure profile start
+    private static final boolean CHAT_SIGNING = Boolean.getBoolean("chat.signing.off");
+    private static MessageSignature signatureWrapper(MessageSignature input) {
+        return CHAT_SIGNING ? input : null;
+    }
+    // Enforce insecure profile end
     public static final StreamCodec<FriendlyByteBuf, ServerboundChatPacket> STREAM_CODEC = Packet.codec(
         ServerboundChatPacket::write, ServerboundChatPacket::new
     );
 
     private ServerboundChatPacket(FriendlyByteBuf buf) {
-        this(buf.readUtf(256), buf.readInstant(), buf.readLong(), buf.readNullable(MessageSignature::read), new LastSeenMessages.Update(buf));
+        this(buf.readUtf(256), buf.readInstant(), buf.readLong(), signatureWrapper(buf.readNullable(MessageSignature::read)), new LastSeenMessages.Update(buf)); // Enforce insecure profile
     }
 
     private void write(FriendlyByteBuf buf) {
         buf.writeUtf(this.message, 256);
         buf.writeInstant(this.timeStamp);
         buf.writeLong(this.salt);
-        buf.writeNullable(this.signature, MessageSignature::write);
+        // Enforce insecure profile start
+        if (CHAT_SIGNING) {
+            buf.writeNullable(null, MessageSignature::write);
+        } else {
+            buf.writeNullable(this.signature, MessageSignature::write);
+        }
+        // Enforce insecure profile end
         this.lastSeenMessages.write(buf);
     }
 
diff --git a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java
index 5705cb920084b775cce4b361683b32c6b6e003ed..9e4a8c06aeaf1f845ecd12a214952dfb46cc0812 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java
@@ -26,6 +26,10 @@ public record ServerboundChatSessionUpdatePacket(RemoteChatSession.Data chatSess
 
     @Override
     public void handle(ServerGamePacketListener listener) {
-        listener.handleChatSessionUpdate(this);
+        // Enforce insecure profile start
+        if (!Boolean.getBoolean("chat.signing.off")) {
+            listener.handleChatSessionUpdate(this);
+        }
+        // Enforce insecure profile end
     }
 }
