From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sun, 28 Jun 2020 19:36:55 -0400
Subject: [PATCH] Don't allow null UUID's for chat


diff --git a/src/main/java/net/minecraft/network/protocol/game/ClientboundChatPacket.java b/src/main/java/net/minecraft/network/protocol/game/ClientboundChatPacket.java
index 71f1fbebcc7bcf2f5a187d913f7708c5a1893bda..7a8abf9d95fe6ad3934993cb2dd9cba5d612b714 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ClientboundChatPacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ClientboundChatPacket.java
@@ -19,7 +19,7 @@ public class ClientboundChatPacket implements Packet<ClientGamePacketListener> {
         this.message = message;
         this.adventure$message = null; // Paper
         this.type = location;
-        this.sender = sender;
+        this.sender = sender != null ? sender : net.minecraft.Util.NIL_UUID;
     }
 
     // Paper start
