From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lordpipe <lordpipe@protonmail.com>
Date: Fri, 1 Jul 2022 19:04:31 -0600
Subject: [PATCH] Make Spigot spam limit customizable


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 630a762b71861bfe21c47a11d4fe05e1a3b7d339..b421a8d59cf05da6f7de0cbd9c82dc42432b0702 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2400,7 +2400,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
         }
         // Spigot end
         // this.chatSpamTickCount += 20;
-        if (this.chatSpamTickCount.addAndGet(20) > 200 && !this.server.getPlayerList().isOp(this.player.getGameProfile())) {
+        if (this.chatSpamTickCount.addAndGet(20) > Integer.getInteger("spigot.spamLimit", 200) && !this.server.getPlayerList().isOp(this.player.getGameProfile())) {
             if (!isSync) {
                 Waitable waitable = new Waitable() {
                     @Override
